xarray/backends/common.py

```diff
@@ -143,28 +144,31 @@ def _find_absolute_paths(
     ['common.py']
     """
     if isinstance(paths, str):
-        if is_remote_uri(paths) and kwargs.get("engine") == "zarr":
-            if TYPE_CHECKING:
-                import fsspec
+        if is_remote_uri(paths):
+            if not is_http_url(paths):
+                # remote uris that are not http are likely s3 or similar, which support globbing with fsspec
+
+                if TYPE_CHECKING:
+                    import fsspec
+                else:
+                    fsspec = attempt_import("fsspec")
+
+                fs, _, _ = fsspec.core.get_fs_token_paths(
+                    paths,
+                    mode="rb",
+                    storage_options=kwargs.get("backend_kwargs", {}).get(
+                        "storage_options", {}
+                    ),
+                    expand=False,
+                )
+                tmp_paths = fs.glob(fs._strip_protocol(paths))  # finds directories
+                return [fs.get_mapper(path) for path in tmp_paths]
```

Comment:
I thought zarr, in particular, didn't wan't mappers any more, but Store objects, or URLs that will be immediately converted to Stores within zarr anyway.

Reply:
@jhamman has the recommendation changed here recently?